### PR TITLE
Fix Another Off By 1 In Our Finalized Trie

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Bug Fix

**What does this PR do? Why is it needed?**

- [x] Use correct eth1 deposit index when inserting into the trie.
- [x] Fix references across the repo

**Which issues(s) does this PR fix?**

N.A

**Other notes for review**

### CODE REVIEW

**Comment by nisdas** [comment]
PR is blocked till we have more data from our cluster.

File: beacon-chain/powchain/service.go [hunk 1]
```diff
@@ -413,11 +413,16 @@ func (s *Service) initDepositCaches(ctx context.Context, ctrs []*ethpb.DepositCo
 		// accumulates. we finalize them here before we are ready to receive a block.
 		// Otherwise, the first few blocks will be slower to compute as we will
 		// hold the lock and be busy finalizing the deposits.
-		s.cfg.depositCache.InsertFinalizedDeposits(ctx, int64(currIndex)) // lint:ignore uintcast -- deposit index will not exceed int64 in your lifetime.
+		//
```
Comment (rkapka) [inline beacon-chain/powchain/service.go:416]:
```suggestion
```

Comment (rkapka) [inline beacon-chain/powchain/service.go:416]:
```suggestion
```

File: beacon-chain/powchain/service.go [hunk 2]
```diff
@@ -413,11 +413,16 @@ func (s *Service) initDepositCaches(ctx context.Context, ctrs []*ethpb.DepositCo
...
+		// The deposit index in the state is always the index of the next deposit
+		// to be included(rather than the last one to be processed). This was most likely
+		// done as the state cannot represent signed integers.
+
```
Comment (rkapka) [inline beacon-chain/powchain/service.go:420]:
```suggestion
```

**Review by rkapka — APPROVED** [review]

**Pull request merged**